astropy/table/tests/test_table.py

```diff
@@ -2208,6 +2208,18 @@ def test_table_meta_copy():
     assert t2.meta[1] is not t.meta[1]  # Value is NOT the same list object
 
 
+def test_init_table_with_names_and_structured_dtype():
+    """Test fix for #10393"""
+    arr = np.ones(2, dtype=np.dtype([('a', 'i4'), ('b', 'f4')]))
+    t1 = Table(arr)
+    for dtype in (arr.dtype, t1.dtype):
+        t2 = Table(names=['x', 'y'], dtype=arr.dtype)
```

Comment:
```suggestion
        t2 = Table(names=['x', 'y'], dtype=dtype)
```
don't you want to iterate over both? ;-)